modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -586,8 +590,175 @@ private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes the given column based on the preferred width of all items contained in it. This can be potentially very
+     * expensive if the number of rows is large.
+     *
+     * @param tableSkin the {@code TableViewSkinBase} used to determine if we are dealing with a {@link TableView} or a
+     *                  {@link TreeTableView}
+     * @param tc        the column to resize
+     * @param maxRows   the number of rows considered when resizing. If -1 is given, all rows are considered.
+     * @since 12
+     */
+    protected void resizeColumnToFitContent(TableViewSkinBase<?,?,?,?,?> tableSkin, TableColumnBase<?, ?> tc, int maxRows) {
```

Comment:
Can you say why `TableViewSkinBase tableSkin` needs to be passed as an argument to this method? Unless I'm missing something, this method is always called with `this.getTableSkin()`, which suggests that the `tableSkin` parameter can be removed.